Libs/RaptureCore/src/main/java/rapture/kernel/BlobApiImpl.java

```diff
@@ -125,14 +125,14 @@ public void deleteBlobRepo(CallingContext context, String blobRepoUri) {
                     }
                 }
             }
-            SearchPublisher.publishDeleteMessage(context, Kernel.getRepoCacheManager().getBlobConfig(uri.getAuthority()), uri);
         } catch (RaptureException e) {
             log.info("Unable to delete children; repo definitian may be invalid");
         }
 
         // delete parent directory
         BlobRepoConfigStorage.deleteByAddress(uri, context.getUser(), "Remove blob repo");
         removeRepoFromCache(uri.getAuthority());
+        SearchPublisher.publishDropMessage(context, uri.toString());
```

Comment:
I'll take a look, but I've tested and included the test cases - see SearchApiIntegrationTest.java lines 152 onward
The code does a search, assert that there's one response. 
Deletes the repo, repeats the search, asserts that there's zero responses.